tls/s2n_crl.c

```diff
@@ -0,0 +1,85 @@
+/*
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include "s2n_crl.h"
+#include "tls/s2n_connection.h"
+
+static struct s2n_x509_crl* s2n_x509_crl_new(void) {
+    DEFER_CLEANUP(struct s2n_blob mem = { 0 }, s2n_free);
+    PTR_GUARD_POSIX(s2n_alloc(&mem, sizeof(struct s2n_x509_crl)));
+
+    struct s2n_x509_crl *crl = (struct s2n_x509_crl*)(void*) mem.data;
+    crl->crl = NULL;
+
```

Comment:
Rather than doing this, I'd suggest just zeroing the struct memory https://github.com/aws/s2n-tls/blob/main/utils/s2n_blob.h#L45. That way you don't need to update this method if you add more fields.

Reply:
That definitely makes sense! I changed this.